### ISSUE TITLE
DNSDB_SERVER parsing maybe should fail if not a URL?

### ISSUE THREAD

**Issue by djw1149** [dnsdb/dnsdbq#23]
`$ DNSDB_SERVER=https://api-fmt.dnsdb.info dnsdbq  ...`
works, but
`$ DNSDB_SERVER=api-fmt.dnsdb.info dnsdbq  ...`
fails with 
```
libcurl: 301 (http://api-fmt.dnsdb.info/lookup/...)
API: <html>
<head><title>301 Moved Permanently</title></head>
<body bgcolor="white">
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx/1.2.1</center>
</body>
</html>
```
because bare hostname is built into a http: URL, not https:.
Maybe the DNSDB_SERVER parsing code should fail if the value is not a URL?

**Comment by vixie** [comment]
djw1149 wrote:
> |$ DNSDB_SERVER=https://api-fmt.dnsdb.info dnsdbq ...|
> works, but
> |$ DNSDB_SERVER=api-fmt.dnsdb.info dnsdbq ...|
> fails with
>
> |libcurl: 301 (http://api-fmt.dnsdb.info/lookup/...)
> API:<html>
> <head><title>301 Moved Permanently</title></head>
> <body bgcolor="white">
> <center><h1>301 Moved Permanently</h1></center>
> <hr><center>nginx/1.2.1</center>
> </body>
> </html>
> |
>
> because bare hostname is built into a http: URL, not https:.

that is standard behaviour for curl, libcurl, and all web browsers.

> Maybe the DNSDB_SERVER parsing code should fail if the value is not a URL?

we aren't parsing it; we're handing it to libcurl. if libcurl has an 
option to tell it to default to https rather than http, we should use 
it. note that this is a multi-provider client; some dnsdb providers (for 
example, cert-at, or qihoo-360) may choose not to use https, in which 
case the current default won't be wrong.

-- 
P Vixie

**Comment by djw1149** [comment]
Conclusion from offline discussion: We will change the code to prepend https:// if there is no :// in the DNSDB_SERVER string.

**Comment by djw1149** [comment]
Fixed in commit 0f0496d3c3d99ff006b627025967739e969e9d09
